### PR TITLE
Return the biostorage CBM

### PR DESCRIPTION
**Pull request by chaisawlajatang**
#### Summary

```SUMMARY: Content "Return the biostorage CBM"```

#### Purpose of change

This CBM was moved to obsolete some time ago https://github.com/CleverRaven/Cataclysm-DDA/pull/30962 , but I believe it deserves a chance to live, and that's why.
 - It offers a unique mechanic (no other bionic increases storage capacity)
 - It has a nice design (a hidden implanted container fits the game concept)
 - It's realistic. The main reason AFAIK was a part of description, mentioning chest cavity. There's absolutely no space in the chest cavity, yes, but an **abdominal** cavity has a lots of it and can stretch a lot (it does every day when you eat, in fact)
 - It's achievable even on our current IRL level of technology

#### Describe the solution

Move the CBM back from the obsolete, change the description a bit.

#### Describe alternatives you've considered

Leave everything as is.

#### Additional context

N/A

### CODE REVIEW

File: data/json/bionics.json [hunk 1]
```diff
@@ -920,6 +920,14 @@
     ],
     "flags": [ "BIONIC_FAULTY" ]
   },
+  {
+    "id": "bio_storage",
+    "type": "bionic",
+    "name": "Internal Storage",
+    "description": "A bioplastic bladder has been surgically implanted inside your abdominal cavity. It has an external opening with a lid, serving as a storage space. You may carry an extra 2 liters of volume.",
```
Comment (AMurkin) [inline data/json/bionics.json]:
```suggestion
    "description": "A bioplastic bladder has been surgically implanted inside your abdominal cavity.  It has an external opening with a lid, serving as a storage space.  You may carry an extra 2 liters of volume.",
```

**Comment by Rail-Runner** [comment]
Well, I think if this is still deemed not realistic enough for mainline, I think it could be in Aftershock.

**Comment by KorGgenT** [comment]
this should not be put into mainline, but you can add it to aftershock if you wish.

**Comment by chaisawlajatang** [comment]
> this should not be put into mainline, but you can add it to aftershock if you wish.

Why?

**Comment by I-am-Erk** [comment]
Where are you fitting two litres of storage and how are you accessing it?

**Comment by kevingranade** [comment]
Because it's not remotely realistic, people don't have 2L of space to spare anywhere in their bodies.

To render this one aspect of it remotely realistic you'd need to add penalties to the survivor similar to those encountered in a late-term pregnancy, which of course means it would be ludicrously impractical and no one would do it.

Additionally there's the question of accessing the contents, and what side effects that would impose on a person by having a new orifice that presumably is subject to sanitary issues.

**Comment by chaisawlajatang** [comment]
But 2 L is not very much, it's like eating 1 L of food and drinking 1 L of water, nowhere near a late-term pregnancy. In a condition known as ascitis abdomen can expand to dozens of liters. And about space to spare - survivor is able to pack himself with kilograms and kilograms of various implants, and required space is already taken into account as slot points.

**Comment by dunstad** [comment]
> It's achievable even on our current IRL level of technology

It might be worth thinking about why nobody does this, then. Mainly, plain ol' bags and clothing with pockets are way better at accomplishing the same thing.

**Comment by Destragon** [comment]
Friends of me had a dog with ascites. Pretty small dog, but somehow it could still fit 1.5 liters of additional water in the abdomen, which was regularly extracted via syringe. It's pretty crazy what can fit into the body. I'm guessing though that the complaint of Kevin&Co is that water, fat, etc manage to fit in very well because they get evenly spread around the body, which is not the same as having a single, concentrated area in your abdomen with a storage unit.
2 liters really doesn't sound that unreasonable though. Why not change the description to make it look more like a kangaroo pouch? Still attached to your body, but tacked on the outside and not implanted internally.

**Comment by I-am-Erk** [comment]
Fitting a hard or semi-hard two liter container inside your body cavity without damaging organs, and without risk of it rupturing and killing you fairly horribly, is a non trivial task. Being able to get in and out of that container without getting infections is an even more non trivial task. Being able to access that container in any useful amount of time while wearing clothing and probably armour is another very non-trivial task.

It is possible to fit a lot of fluid in the abdomen... distributing it through spaces in organs, often at the cost of damage to those organs. A hard 2L storage space is *possible* but would be extremely unwieldy and risky. 

Perhaps most to the point, all these issues were brought up last time and instead of trying to fix them and produce a version that might get around these problems, it seems you just waited until you forgot we'd already had this conversation.

**Comment by chaisawlajatang** [comment]
> Perhaps most to the point, all these issues were brought up last time and instead of trying to fix them and produce a version that might get around these problems, it seems you just waited until you forgot we'd already had this conversation.

Conversation? What conversation? There was NO conversation, there was just "this is silly, removed it" type of thing. Issues? There was no pointed out issues, just "This is the last truly atrociously WTF CBM". 

The same thing is now - no chance of discussion or fixing stuff. Simply closing it up.

**Comment by Rail-Runner** [comment]
Maybe just reduce provided storage volume by a lot? So it's something of a small pocket that is designed for smuggling small objects. Would not be very useful for a survivor, but not all CBMs have to be.

**Comment by Night-Pryanik** [comment]
Once again, how things are put into and taken out of this storage?

**Comment by chaisawlajatang** [comment]
> Once again, how things are put into and taken out of this storage?

There could be a small opening in the lower back or abdominal wall, like [colostomy](https://www.google.ru/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&ved=2ahUKEwjrhPDpxpPlAhWD4aYKHbRpD0gQjRx6BAgBEAQ&url=https%3A%2F%2Fwww.ostomy.org%2Fknow-your-ostomy%2F&psig=AOvVaw2oaycQSbJRn7VVlbKqJ0f-&ust=1570860825768249), but with a bioplastic bladder behind it and a lid to close it.

> Maybe just reduce provided storage volume by a lot?

Sure thing, this could be an option. Like 1 L or 500 mL.

> Fitting a hard or semi-hard two liter container inside your body cavity without damaging organs, and without risk of it rupturing and killing you fairly horribly, is a non trivial task

I can say the same thing for almost all CBMs. Fuel cell claims to have "diffuse network of bio-plastic bladders able to hold up to 500 ml of gasoline", but this is ok apparently. Yeah, it was said several times that one "silly" thing is not a justification for another (though I don't consider biostorage silly), but thing currently ingame serve kinda waymarks or guides to what's ok and what is not.

> Being able to get in and out of that container without getting infections is an even more non trivial task

I imagine the container to be connected directly to the outside via the opening with a lid. It's kinda immobile and attached to the abdominal wall.

> would be extremely unwieldy and risky, generally mind-bogglingly stupid for the purpose being discussed

> It might be worth thinking about why nobody does this, then. Mainly, plain ol' bags and clothing with pockets are way better at accomplishing the same thing

I can say the same thing for almost all CBMs.

**Comment by GroeneAppel** [comment]
Perhaps the solution could be in the trend of making it a 'soft' storage and only allowing liquids in it. This deals with any arguments of it being capable of damaging your organs, the issue of getting the liquid out (pump system/contraction/taking a silly pose), while giving the player a small storage of a liquid of their choice. If needed the amount it can hold can be reduced from anywhere to 1,5L to 500ML (or less). This in addition deals with the size argument as a 500ML moldable container is sure to fit within the body without causing harm as it does not compress the organs much.

**Pull request closed without merging**